sql/src/main/java/io/crate/planner/node/management/ExplainPlan.java

```diff
@@ -167,18 +173,83 @@ private TransportCollectProfileOperation getRemoteCollectOperation(DependencyCar
         return new TransportCollectProfileOperation(nodeAction, jobId);
     }
 
-    private Row buildResponse(Map<String, Object> apeTimings, Map<String, Map<String, Object>> nodeTimings) {
+    private Row buildResponse(Map<String, Object> apeTimings,
+                              Map<String, Map<String, Object>> nodeTimings,
+                              NodeOperationTree operationTree) {
         MapBuilder<String, Object> mapBuilder = MapBuilder.newMapBuilder();
         apeTimings.forEach(mapBuilder::put);
 
+        Map<String, Object> phasesTimings = extractPhasesTimingsFrom(nodeTimings, operationTree);
+        Map<String, Map<String, Object>> resultNodeTimings = getNodeTimingsWithoutPhases(phasesTimings.keySet(), nodeTimings);
         MapBuilder<String, Object> executionTimingsMap = MapBuilder.newMapBuilder();
-        nodeTimings.forEach(executionTimingsMap::put);
+        executionTimingsMap.put("Phases", phasesTimings);
+        resultNodeTimings.forEach(executionTimingsMap::put);
         executionTimingsMap.put("Total", apeTimings.get(Phase.Execute.name()));
 
         mapBuilder.put(Phase.Execute.name(), executionTimingsMap.immutableMap());
         return new Row1(mapBuilder.immutableMap());
     }
 
+    private static Map<String, Object> extractPhasesTimingsFrom(Map<String, Map<String, Object>> nodesTimings,
+                                                                NodeOperationTree operationTree) {
+        Map<String, Object> allPhases = new TreeMap<>();
+        for (NodeOperation operation : operationTree.nodeOperations()) {
+            ExecutionPhase phase = operation.executionPhase();
+            String phaseName = getPhaseName(phase);
+            Map<String, Object> phaseTimingsAcrossNodes = getPhaseTimingsAcrossNodes(phaseName, nodesTimings);
+
+            if (!phaseTimingsAcrossNodes.isEmpty()) {
+                allPhases.put(phaseName, MapBuilder.newMapBuilder().put("nodes", phaseTimingsAcrossNodes).immutableMap());
+            }
+        }
+
+        ExecutionPhase leafExecutionPhase = operationTree.leaf();
+        String phaseName = getPhaseName(leafExecutionPhase);
```

Comment:
L207-212 could be extracted into a separate method and reused also in the loop above.